### PR TITLE
Give ownership of Openhab folders to "openhab"

### PR DESCRIPTION
**Pull request by EliasGabrielsson**
Fixes: #534

Signed-off-by: Elias Gabrielsson <elias@gabson.se>

### CODE REVIEW

**Comment by EliasGabrielsson** [comment]
Help me think this through. As the user `openhabian` have access to the group `openhab` this ownership change should not make any issues right?

**Comment by EliasGabrielsson** [comment]
I also wonder if we shall only make the folders and the config files readable by `openhab` and `openhabian` user?

These files can contain quite sensitive API keys etc. 
Does any other service then openhab (+ the openhabian user) needs to have access to the folders? 

Fronttrail what permission is that service using?

**Comment by mstormi** [comment]
I'm a little confused. Changing permissions is a sensitive thing that easily can go wrong.
A (first, incomplete) number of questions to come to mind:
* Do we always have a openhabian user ? What if someone's installing but not using the image ?
* we mustn't expect users to use the openhab user for interactive works. It used to be "openhabian" or something else but not "openhab". 
* Your patch shows .... `chown -R openhab:$username` ... why, use the USER name for the GROUP ?

**Comment by EliasGabrielsson** [comment]
> Do we always have a openhabian user ? What if someone's installing but not using the image ?
Yes, we always have a user as we ask for that in non-openhabian installations.

> we mustn't expect users to use the openhab user for interactive works. It used to be "openhabian" or something else but not "openhab".
Sorry I don't get what you mean. The patch changes ownership of openhab runtime folder etc. to the service account. Could you elaborate a bit more? 

> Your patch shows .... chown -R openhab:$username ... why, use the USER name for the GROUP ?
I think there is an "openhabian" group as well but this is clearly confusing. This would also be broken in the case of installing on a non-openhabian installation. I will look into it.

**Comment by mstormi** [comment]
> The patch changes ownership of openhab runtime folder etc. to the service account. Could you elaborate a bit more?

openhab is the account OH runs as, but a user does not necessarily use it to configure it. That's "openhabian", "pi" or even something else.
We need files to be writeable for the OH group and "openhab" and (all) the service account(s) need to be members of that group as well.

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ node_modules/
 *.img.xz
 *.img.bz2
 *.log
+.DS_Store
```
Comment (ThomDietrich) [inline .gitignore]: I stopped to add these files to every repository individually and added them to my systems global gitignore instead. Still not against this, but maybe at all while at it? https://github.com/ebeshero/DHClass-Hub/blob/master/.gitignore

File: functions/system.bash [hunk 2]
```diff
@@ -176,11 +176,11 @@ permissions_corrections() {
...
 
   #
-  openhab_folders=(/etc/openhab2 /var/lib/openhab2 /var/log/openhab2 /usr/share/openhab2/addons)
+  openhab_folders=(/etc/openhab2 /var/lib/openhab2 /var/log/openhab2 /usr/share/openhab2)
   cond_redirect chown openhab:$username /srv /srv/README.txt
   cond_redirect chmod ugo+w /srv /srv/README.txt
-  cond_redirect chown -R openhab:openhab /usr/share/openhab2
-  cond_redirect chown -R openhab:$username /opt ${openhab_folders[@]}
+  cond_redirect chown -R openhab:openhab /usr/share/openhab2 ${openhab_folders[@]}
+  cond_redirect chown -R openhab:$username /opt
```
Comment (ThomDietrich) [inline functions/system.bash]: I agree with @BClark09. Just merge both lines for `openhab:openhab` and get rid of all other `openhab:$username`.

**Comment by EliasGabrielsson** [comment]
Take a look at this please. Are we settled?

**Comment by mstormi** [comment]
There's a couple of places where files in the provisioning user's homedir are set group ownership $username like
functions/openhab.bash:114:  chown -R "$username:$username" /home/$username/.vim
functions/system.bash:213:  chown -R "$username:$username" /home/"$username"/.ssh
functions/system.bash:193:  cond_redirect chown -R "$username:$username /home/$username"

Would we want to keep that ? or change to group openhab ?

Working on PR #613 there's also a couple of places shellcheck marked where the $username variable eventually isn't set and I changed those to have a default now.
Wasn't always sure what to insert, e.g. the user and group to use when installing node-red ?
Maybe worth double checking those places for this issue here.

**Comment by EliasGabrielsson** [comment]
Hmm, the above cases regards the homefolder so we shouldn't mess with a service account over there.

**Pull request merged**